src/build_options.rs

```diff
@@ -382,44 +388,44 @@ fn has_abi3(cargo_metadata: &Metadata) -> Result<Option<(u8, u8)>> {
         .resolve
         .as_ref()
         .context("Expected cargo to return metadata with resolve")?;
-    let pyo3_packages = resolve
-        .nodes
-        .iter()
-        .filter(|package| cargo_metadata[&package.id].name == "pyo3")
-        .collect::<Vec<_>>();
-    match pyo3_packages.as_slice() {
-        [pyo3_crate] => {
-            // Find the minimal abi3 python version. If there is none, abi3 hasn't been selected
-            // This parser abi3-py{major}{minor} and returns the minimal (major, minor) tuple
-            let abi3_selected = pyo3_crate.features.iter().any(|x| x == "abi3");
-
-            let min_abi3_version = pyo3_crate
-                .features
-                .iter()
-                .filter(|x| x.starts_with("abi3-py") && x.len() >= "abi3-pyxx".len())
-                .map(|x| {
-                    Ok((
-                        (x.as_bytes()[7] as char).to_string().parse::<u8>()?,
-                        x[8..].parse::<u8>()?,
-                    ))
-                })
-                .collect::<Result<Vec<(u8, u8)>>>()
-                .context("Bogus pyo3 cargo features")?
-                .into_iter()
-                .min();
-            if abi3_selected && min_abi3_version.is_none() {
-                bail!(
-                    "You have selected the `abi3` feature but not a minimum version (e.g. the `abi3-py36` feature). \
-                    maturin needs a minimum version feature to build abi3 wheels."
-                )
+    for &lib in PYO3_BINDING_CRATES.iter() {
+        let pyo3_packages = resolve
+            .nodes
+            .iter()
+            .filter(|package| cargo_metadata[&package.id].name.as_str() == lib)
+            .collect::<Vec<_>>();
+        match pyo3_packages.as_slice() {
+            [pyo3_crate] => {
+                // Find the minimal abi3 python version. If there is none, abi3 hasn't been selected
+                // This parser abi3-py{major}{minor} and returns the minimal (major, minor) tuple
+                let abi3_selected = pyo3_crate.features.iter().any(|x| x == "abi3");
```

Comment:
What if both `pyo3-ffi` and `pyo3` are listed in dependencies? For example

```toml
[dependencies]
pyo3 = { version = "0.16", features = ["abi3"] }
pyo3-ffi = "0.16"
```

Since `pyo3-ffi` is tried first, it won't see the `abi3` feature from `pyo3`.

Reply:
I think this is fine since we're getting features from `cargo metadata`.